lib/plugins/aws/package/compile/events/alexaSkill/index.js

```diff
@@ -28,34 +38,16 @@ class AwsCompileAlexaSkillEvents {
             let enabled = true;
             let appId;
             if (event === 'alexaSkill') {
-              const warningMessage = [
-                "Warning! You are using an old syntax for alexaSkill which doesn't",
-                ' restrict the invocation solely to your skill.',
-                ' Please refer to the documentation for additional information.',
-              ].join('');
-              this.serverless.cli.log(warningMessage);
+              this.serverless._logDeprecation(
+                'ALEXA_SKILL_EVENT_WITHOUT_APP_ID',
+                'Starting with next major version, support for alexaSkill event without appId specified will be removed.'
+              );
             } else if (typeof event.alexaSkill === 'string') {
               appId = event.alexaSkill;
-            } else if (_.isPlainObject(event.alexaSkill)) {
-              if (typeof event.alexaSkill.appId !== 'string') {
-                const errorMessage = [
-                  `Missing "appId" property for alexaSkill event in function ${functionName}`,
-                  ' The correct syntax is: appId: amzn1.ask.skill.xx-xx-xx-xx-xx',
-                  ' OR an object with "appId" property.',
-                  ' Please check the docs for more info.',
-                ].join('');
-                throw new this.serverless.classes.Error(errorMessage);
-              }
+            } else if (typeof event.alexaSkill === 'object') {
```

Comment:
As it's a last acceptable form, we may simplify this to `} else {` (it'll be nice also because `typeof x === 'object'` feels quirky (e.g. `null` passes such condition))

Reply:
good call, changed :+1: